app/rtc_compare.py

```diff
@@ -0,0 +1,488 @@
+'''
+Compare two RTC products (in HDF5) if they are equivalent.
+Part of the codes are copied from PROTEUS SAS
+'''
+
+import argparse
+import itertools
+
+import h5py
+import numpy as np
+
+RTC_S1_PRODUCTS_ERROR_TOLERANCE = 1e-6
+
+def _get_parser():
+    parser = argparse.ArgumentParser(
+        description='Compare two RTC products',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    # Inputs
+    parser.add_argument('input_file',
+                        type=str,
+                        nargs=2,
+                        help='Input RTC products in NETCDF/HDF5 format')
+
+    return parser
+
+
+def _unpack_array(val_in, hdf5_obj_in):
+    '''
+    Unpack the array of array into ordinary numpy array.
+    Convert an HDF5 object reference into the path it is pointing to.
+
+    For internal use in this script.
+
+    Parameter:
+    -----------
+    val_in: np.ndarray
+        numpy array to unpack
+    hdf5_obj_in:
+        Source HDF5 object of `val_in`
+
+    Return:
+    val_out: np.ndarray
+        unpacked array
+
+    '''
+    list_val_in = list(itertools.chain.from_iterable(val_in))
+
+    list_val_out = [None] * len(list_val_in)
+    for i_val, element_in in enumerate(list_val_in):
+        if isinstance(element_in, h5py.h5r.Reference):
+            list_val_out[i_val] = np.str_(hdf5_obj_in[element_in].name)
+        else:
+            list_val_out[i_val] = element_in
+    val_out = np.array(list_val_out)
+
+    return val_out
+
+
+def print_data_difference(val_1, val_2, indent=4):
+    '''
+    Print out the difference of the data whose dimension is >= 1
+
+    Parameters:
+    -----------
+    val_1, val_2: np.array
+        Data that has difference to each other
+    indent: int
+        Number of spaces for indentation
+    '''
+
+    str_indent = ' ' * indent + '-'
+
+    # printout the difference
+    if not issubclass(val_1.dtype.type, np.number):
+        # Routine for non-numeric array
+        # Print out the first discrepancy in the array
+        flag_discrepancy = (val_1 != val_2)
+        index_first_discrepancy = np.where(flag_discrepancy)[0][0]
+        print(f'{str_indent} The first discrepancy has detected from index '
+              f'[{index_first_discrepancy}] : '
+              f'1st=({val_1[index_first_discrepancy]}), '
+              f'2nd=({val_2[index_first_discrepancy]})')
+        return None
```

Comment:
```suggestion
        return
```

`return None` is the same as `return`